consensus/hotstuff/votes_queue.go

```diff
@@ -0,0 +1,22 @@
+package hotstuff
+
+import "github.com/onflow/flow-go/consensus/hotstuff/model"
+
+// VotesQueue is a queue to buffer the votes to be processed by multiple workers concurrently.
+type VotesQueue interface {
```

Comment:
as I said previously, I am really not sure how much value we have from `VotesQueue`, I would simply implement queuing as part of `VoteAggregator`

Reply:
I'm trying to abstract easy layer, making each layer taking less responsibility, but I think your approach also works. 

I think we can still keep this interface here, it won't create merge conflicts. If in the end, we still think this abstraction is unnecessary, we can remove it.

Reply:
If you think of every struct you can put signature material in as a vertex, you can represent the data flow of signature material this way (letters arbitrary):

```
A -> B -> C -> D -> E -> F
```

Let's say F is your final state (a QC object, fully aggregated and verified), and C is where each signature's verification occurs.

What you do not want to see happen (and what auditors will look for) is this:

```
A -> B -> C(*) -> D -> E -> F
                            ^
                           /
                          G
```

In `G` somebody introduces signature material into the system, by passing validation (at *)

One chain that makes it less likely for that to happen is the following:
```
A -> B -> H
```

Where `H` combines `C -> D -> E -> F` in a single component. A Data flow such as :
```
A -> B  -> C(*) -> I
```

Where `I` combines `D -> E -> F ` is not as good , but still better than the original.

In other terms: We want concurrency in verification plus double-purpose accounting of beacon signatures. That absolutely forces a few steps in our vote processing data flow. But if we minimize further intermediate states as much as possible, I think we'll all have a better time verifying we're not leaving gaps in the state machine.

Reply:
Agreed. but not sure if you are suggest any changes to this module?